x/mongo/driver/auth/mongodbaws.go

```diff
@@ -34,6 +37,20 @@ type MongoDBAWSAuthenticator struct {
 	sessionToken string
 }
 
+type mongoDBAWSAuthenticator struct {
+	creds.AwsCredentials
+	creds.AwsCredentialProvider
+}
+
+// GetCredentials gets AWS credentials.
+func (a *mongoDBAWSAuthenticator) GetCredentials(ctx context.Context) (*awsv4.StaticProvider, error) {
```

Comment:
```suggestion
func (a *mongoDBAWSAuthenticator) getCredentials(ctx context.Context) (*awsv4.StaticProvider, error) {
```

Suggest not exporting unless this may be needed outside the `auth` package.